### PR TITLE
CR-2158: [Not CAR] [AWS] Ensure ALB drops invalid HTTP headers

### PR DESCRIPTION
**Pull request by Imanuel-Mizrahi**
created new resource, added tests and rule

### CODE REVIEW

File: cloudrail/knowledge/context/aws/elb/load_balancer.py [hunk 1]
```diff
@@ -51,8 +52,8 @@ def __init__(self, account: str, region: str, name: str, scheme_type: LoadBalanc
         self.load_balancer_arn: str = load_balancer_arn
         self.target_groups: List[LoadBalancerTargetGroup] = []
         self.listener_ports: List[int] = []
-
         self.raw_data = LoadBalancerRawData()
+        self.load_balancer_attributes: Optional[LoadBalancerAttributes] = None
```
Comment (yi2020) [inline cloudrail/knowledge/context/aws/elb/load_balancer.py:56]: Missing documentation

File: cloudrail/knowledge/context/aws/elb/load_balancer_attributes.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from typing import List
```
Comment (yi2020) [inline cloudrail/knowledge/context/aws/elb/load_balancer_attributes.py:1]: This file needs to be added to the elb readme in the docs

File: cloudrail/knowledge/rules/aws/non_context_aware/protocol_enforcments/ensure_load_balancer_drops_invalid_http_headers_rule.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+class EnsureLoadBalancerDropsInvalidHttpHeadersRule(AwsBaseRule):
+
+    def get_id(self) -> str:
+        return 'non_car_alb_drops_invalid_http_headers'
+
+    def execute(self, env_context: EnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+
+        for lb in env_context.load_balancers:
+            if lb.load_balancer_attributes and not lb.load_balancer_attributes.drop_invalid_header_fields:
```
Comment (yi2020) [inline cloudrail/knowledge/rules/aws/non_context_aware/protocol_enforcments/ensure_load_balancer_drops_invalid_http_headers_rule.py:18]: Check if the lb is application-lb and not nlb.

**Review by yi2020 — CHANGES_REQUESTED** [review]

File: cloudrail/knowledge/rules/aws/non_context_aware/protocol_enforcments/ensure_load_balancer_drops_invalid_http_headers_rule.py (hunk 3 above)
Comment (Imanuel-Mizrahi) [reply cloudrail/knowledge/rules/aws/non_context_aware/protocol_enforcments/ensure_load_balancer_drops_invalid_http_headers_rule.py]: Code changed, @yi2020 , please review.
Thanks!

**Review by yi2020 — APPROVED** [review]

**Pull request merged**